47.12 Stock save games will not load

47.12
Linux

Fresh install with no modifications or alterations
 
"Load Last Save' From the boot screen or attempting to load the stock save games from the Campaign folder produce campaign loading error popup. Selecting 'Start New Campaign' loads without issue. 

[mekhqlog.txt](https://github.com/MegaMek/mekhq/files/5508076/mekhqlog.txt)

Same issue here.  

[Onderon's Outlaws30671220_save.cpnx.gz](https://github.com/MegaMek/mekhq/files/5508100/Onderon.s.Outlaws30671220_save.cpnx.gz)
[megameklog.txt](https://github.com/MegaMek/mekhq/files/5508101/megameklog.txt)
[mekhqlog.txt](https://github.com/MegaMek/mekhq/files/5508102/mekhqlog.txt)
21:34:42,271 INFO [mekhq.campaign.io.CampaignXmlParser] {SwingWorker-pool-1-thread-1} 
processPartNodes(), line 1411 : Loading Part Nodes from XML...

21:34:42,290 ERROR [mekhq.campaign.parts.Part] {SwingWorker-pool-1-thread-1} 
generateInstanceFromXML(), line 860 : null
java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
	at java.lang.reflect.Constructor.newInstance(Unknown Source)
	at mekhq.campaign.parts.Part.generateInstanceFromXML(Part.java:784)
	at mekhq.campaign.io.CampaignXmlParser.processPartNodes(CampaignXmlParser.java:1433)
	at mekhq.campaign.io.CampaignXmlParser.parse(CampaignXmlParser.java:261)
	at mekhq.campaign.CampaignFactory.createCampaign(CampaignFactory.java:83)
	at mekhq.gui.dialog.DataLoadingDialog$Task.doInBackground(DataLoadingDialog.java:211)
	at mekhq.gui.dialog.DataLoadingDialog$Task.doInBackground(DataLoadingDialog.java:123)
	at javax.swing.SwingWorker$1.call(Unknown Source)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at javax.swing.SwingWorker.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
Caused by: java.lang.NullPointerException
	at mekhq.campaign.parts.AmmoStorage.<init>(AmmoStorage.java:61)
	at mekhq.campaign.parts.AmmoStorage.<init>(AmmoStorage.java:55)
	... 16 more

--- a/MekHQ/src/mekhq/campaign/parts/AmmoStorage.java
+++ b/MekHQ/src/mekhq/campaign/parts/AmmoStorage.java
@@ -33,6 +33,7 @@ import megamek.common.EquipmentType;
 import megamek.common.ITechnology;
 import megamek.common.TargetRoll;
 import megamek.common.TechAdvancement;
+import megamek.common.annotations.Nullable;
 import mekhq.MekHqXmlUtil;
 import mekhq.campaign.Campaign;
 import mekhq.campaign.parts.equipment.EquipmentPart;
@@ -55,10 +56,12 @@ public class AmmoStorage extends EquipmentPart implements IAcquisitionWork {
     	this(0, null, 0, null);
     }
 
-    public AmmoStorage(int tonnage, EquipmentType et, int shots, Campaign c) {
+    public AmmoStorage(int tonnage, @Nullable EquipmentType et, int shots, @Nullable Campaign c) {
         super(tonnage, et, -1, 1.0, c);
         this.shots = shots;
-        this.munition = getType().getMunitionType();
+        if (et instanceof AmmoType) {
+            this.munition = ((AmmoType) et).getMunitionType();
+        }
     }
 
     public AmmoStorage clone() {

--- a/MekHQ/unittests/mekhq/campaign/parts/AmmoStorageTest.java
+++ b/MekHQ/unittests/mekhq/campaign/parts/AmmoStorageTest.java
@@ -1,0 +1,32 @@
+/*
+ * Copyright (C) 2020 MegaMek team
+ *
+ * This file is part of MekHQ.
+ *
+ * MekHQ is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MekHQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MekHQ.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package mekhq.campaign.parts;
+
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+
+public class AmmoStorageTest {
+    @Test
+    public void ammoStorageDeserializationCtorTest() {
+        AmmoStorage ammoStorage = new AmmoStorage();
+        assertNotNull(ammoStorage);
+    }
+}

